Problem Statement 1:
In each of the following situations, state whether it is a correctly stated hypothesis
testing problem and why?
1. 𝐻0: 𝜇 = 25, 𝐻1: 𝜇 ≠ 25
2. 𝐻0: 𝜎 > 10, 𝐻1: 𝜎 = 10
3. 𝐻0: 𝑥 = 50, 𝐻1: 𝑥 ≠ 50
4. 𝐻0: 𝑝 = 0.1, 𝐻1: 𝑝 = 0.5
5. 𝐻0: 𝑠 = 30, 𝐻1: 𝑠 > 30

When writing hypotheses there are three things that we need to know: 
(1) the parameter that we are testing -->single mean(𝜇), paired means(𝜇_d), a single proportion(𝑝), the difference between two independent means(𝜇_1-𝜇_2), the difference between two proportions(𝑝_1-𝑝_2), a simple linear regression slope(Beta), and a correlation(rho). 
(2) the direction of the test (non-directional, right-tailed or left-tailed)-->research will give us direction
(3) the value of the hypothesized parameter-->research will give us the value

1. 𝐻0: 𝜇 = 25, 𝐻1: 𝜇 ≠ 25--> Hypothesis correctly stated.
2. 𝐻0: 𝜎 > 10, 𝐻1: 𝜎 = 10--> Hypothesis correctly stated.
3. 𝐻0: 𝑥 = 50, 𝐻1: 𝑥 ≠ 50--> Hypothesis not correct, since not made on a sample.
4. 𝐻0: 𝑝 = 0.1, 𝐻1: 𝑝 = 0.5--> Hypothesis correctly stated.
5. 𝐻0: 𝑠 = 30, 𝐻1: 𝑠 > 30--> Hypothesis not correct, since not made on a sample.

Problem Statement 2:
The college bookstore tells prospective students that the average cost of its
textbooks is Rs. 52 with a standard deviation of Rs. 4.50. A group of smart statistics
students thinks that the average cost is higher. To test the bookstore’s claim against
their alternative, the students will select a random sample of size 100. Assume that
the mean from their random sample is Rs. 52.80. Perform a hypothesis test at the
5% level of significance and state your decision.
H0: mu=52
H1: mu!=52

H0: mu=52
H1: mu!=52

In [1]:
from scipy import stats
import numpy as np

sigma,n,mu,x_mean = 4.5,100,52,52.8
z_test = (x_mean-mu)/(sigma/np.sqrt(n))
z_test

1.7777777777777715

In [2]:
alpha = 0.05
z_critical=stats.norm.ppf(1-alpha/2) #Two-tail test
z_critical

1.959963984540054

In [3]:
print("H0 rejected") if z_test > z_critical else print("H0 accepted")

H0 accepted


Problem Statement 3:
A certain chemical pollutant in the Genesee River has been constant for several
years with mean μ = 34 ppm (parts per million) and standard deviation σ = 8 ppm. A
group of factory representatives whose companies discharge liquids into the river is
now claiming that they have lowered the average with improved filtration devices. A
group of environmentalists will test to see if this is true at the 1% level of
significance. Assume that their sample of size 50 gives a mean of 32.5 ppm.
Perform a hypothesis test at the 1% level of significance and state your decision.

In [4]:
H0: mu = 34
H1:mu != 34

In [5]:
from scipy import stats
import numpy as np

sigma,n,mu,x_mean = 8,50,34,32.5
z_test = (x_mean-mu)/(sigma/np.sqrt(n))
z_test

-1.3258252147247767

In [6]:
alpha = 0.01
z_critical=stats.norm.ppf(1-alpha/2) #Two-tail test
z_critical

2.5758293035489004

In [7]:
print("H0 rejected") if z_test > z_critical else print("H0 accepted")

H0 accepted


Problem Statement 4:
Based on population figures and other general information on the U.S. population,
suppose it has been estimated that, on average, a family of four in the U.S. spends
about $1135 annually on dental expenditures. Suppose further that a regional dental
association wants to test to determine if this figure is accurate for their area of
country. To test this, 22 families of 4 are randomly selected from the population in
that area of the country and a log is kept of the family’s dental expenditure for one
year. The resulting data are given below. Assuming, that dental expenditure is
normally distributed in the population, use the data and an alpha of 0.05 to test the
dental association’s hypothesis.
x=[1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699,
872, 913, 944, 954, 987, 1695, 995, 1003, 994]

H0: mu = 1135
H1: mu != 1135

In [8]:
x=[1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699, 872, 913, 944, 954, 987, 1695, 995, 1003, 994]

In [9]:
from scipy import stats
import numpy as np

#Since n<30, we use ttest
alpha = 0.05
mu = 1165
t_value,p_value = stats.ttest_1samp(x,mu)
print(p_value)

0.016406635559183057


In [10]:
print("H0 rejected") if p_value < alpha else print("H0 accepted")

H0 rejected


Problem Statement 5:
In a report prepared by the Economic Research Department of a major bank the
Department manager maintains that the average annual family income on Metropolis
is $48,432. What do you conclude about the validity of the report if a random sample
of 400 families shows and average income of $48,574 with a standard deviation of
2000?

H0 : mu = 48432
H1 : mu != 48432

In [11]:
from scipy import stats
import numpy as np

sigma,n,mu,x_mean = 2000,400,48432,48574
z_test = (x_mean-mu)/(sigma/np.sqrt(n))
z_test

1.42

In [12]:
#Assume alpha=5%
alpha = 0.05
z_critical=stats.norm.ppf(1-alpha/2) #Two-tail test
z_critical

1.959963984540054

In [13]:
print("H0 rejected") if z_test > z_critical else print("H0 accepted")

H0 accepted


Problem Statement 6:
Suppose that in past years the average price per square foot for warehouses in the
United States has been $32.28. A national real estate investor wants to determine
whether that figure has changed now. The investor hires a researcher who randomly
samples 19 warehouses that are for sale across the United States and finds that the
mean price per square foot is $31.67, with a standard deviation of $1.29. assume
that the prices of warehouse footage are normally distributed in population. If the
researcher uses a 5% level of significance, what statistical conclusion can be
reached? What are the hypotheses?

H0: mu >=32.28
H1: mu<32.28

In [14]:
from scipy import stats
import numpy as np

#n<30 use t-test

sigma,n,mu,x_mean = 1.29,19,32.28,31.67
t_test = (x_mean-mu)/(sigma/np.sqrt(n))
t_test

-2.06118477175179

In [15]:
alpha = 0.05
p = 1-alpha
df = n-1
t_critical = stats.t.ppf(p,df)
t_critical

1.7340636066175354

In [16]:
print("H0 rejected") if t_test > t_critical else print("H0 accepted")

H0 accepted


Problem Statement 7:
Fill in the blank spaces in the table and draw your conclusions from it.
Acceptance         (Sample Size) α     (β= at 𝜇 = 52) (β= at 𝜇 = 50.5)    Conclusion_Mu=52         Conclusion_mu=50.5
48.5 < x̄ < 51.5     10           0.05         0.0011	0.1605             Reject Null Hypothesis	Accept Null Hypothesis
48 < x̄ < 52         10           0.05         0.0058	0.2249             Reject Null Hypothesis	Accept Null Hypothesis
48.81 < x̄ < 51.9    16           0.05         0.0041	0.4884             Reject Null Hypothesis	Accept Null Hypothesis
48.42 < x̄ < 51.58   16           0.05         0.0016	0.1717             Reject Null Hypothesis	Accept Null Hypothesis

x_accpt_lower	x_accpt_upper	x_mean	s	n	mu_1	mu_2	t_1_test	t_2_test	Beta_1	Beta_2	alpha	t_alpha	Conclusion_mu=52	Conclusion_mu=50.5
48.50	51.50	50.00	1.50	10.00	52.00	50.50	-4.22	-1.05	0.0011	0.1605	0.05	1.73	Reject Null Hypothesis	Accept Null Hypothesis
48.00	52.00	50.00	2.00	10.00	52.00	50.50	-3.16	-0.79	0.0058	0.2249	0.05	1.73	Reject Null Hypothesis	Accept Null Hypothesis
48.81	51.90	50.36	1.55	10.00	52.00	50.50	-3.37	-0.30	0.0041	0.4884	0.05	1.73	Reject Null Hypothesis	Accept Null Hypothesis
48.42	51.58	50.00	1.58	10.00	52.00	50.50	-4.00	-1.00	0.0016	0.1717	0.05	1.73	Reject Null Hypothesis	Accept Null Hypothesis

Problem Statement 8:
Find the t-score for a sample size of 16 taken from a population with mean 10 when
the sample mean is 12 and the sample standard deviation is 1.5.

In [17]:
from scipy import stats
import numpy as np

#n<30 use t-test

sigma,n,mu,x_mean = 1.5,16,10,12
t_test = (x_mean-mu)/(sigma/np.sqrt(n))
t_test

5.333333333333333

Problem Statement 9:
Find the t-score below which we can expect 99% of sample means will fall if samples
of size 16 are taken from a normally distributed population.

In [18]:
alpha=0.01
p = 1-alpha
n = 16
df = n-1
t_score = stats.t.ppf(p,df)
t_score

2.602480294995493

Problem Statement 10:
If a random sample of size 25 drawn from a normal population gives a mean of 60
and a standard deviation of 4, find the range of t-scores where we can expect to find
the middle 95% of all sample means. Compute the probability that (−𝑡0.05 <𝑡<𝑡0.10).

In [19]:
n = 25
sam_mean = 60
s = 4
p_alpha =0.95
alpha = 1-p_alpha
p = 1-(alpha)/2

lv = sam_mean - stats.t.ppf(p,n-1)*(s/np.sqrt(n))
hv = sam_mean + stats.t.ppf(p,n-1)*(s/np.sqrt(n))
ls = (sam_mean-lv)/(s/np.sqrt(n))
hs = (sam_mean-hv)/(s/np.sqrt(n))
[ls,hs]

[2.063898561628017, -2.063898561628017]

In [20]:
#p(-t0.05<t<t0.1)
prob = (0.5-0.05)+(0.05-0.1)
prob

0.4

Problem Statement 11:
Two-tailed test for difference between two population means
Is there evidence to conclude that the number of people travelling from Bangalore to
Chennai is different from the number of people travelling from Bangalore to Hosur in
a week, given the following:
Population 1: Bangalore to Chennai n1 = 1200
x1 = 452
s1 = 212
Population 2: Bangalore to Hosur n2 = 800
x2 = 523
s2 = 185

H0: mu1-mu2 =0
H1: mu1-mu2 != 0

In [21]:
#using t-test assuming each population is at least 20 times larger than its respective sample.
n1= 1200
x1 =452
s1=212

n2=800
x2=523
s2=185

se = np.sqrt(s1**2/n1 +s2**2/n2)
t_test = (x1-x2)/se
t_test

-7.926428526759299

In [22]:
df = (s1**2/n1 + s2**2/n2)**2 / ( (s1**2 / n1)**2 / (n1 - 1)  + (s2**2 / n2)**2 / (n2 - 1) )
df

1860.2560592148861

In [23]:
#Assuming alpha=0.05
alpha=0.05
p = 1-alpha/2
n = 16
t_score = stats.t.ppf(p,df)
t_score

1.9612400396405612

In [24]:
print("H0 rejected") if t_test > t_critical else print("H0 accepted")

H0 accepted


Problem Statement 12:
Is there evidence to conclude that the number of people preferring Duracell battery is
different from the number of people preferring Energizer battery, given the following:
Population 1: Duracell
n1 = 100
x1 = 308
s1 = 84
Population 2: Energizer
n2 = 100
x2 = 254
s2 = 67

H0: mu1-mu2 =0
H1: mu1-mu2 != 0

In [33]:
#using t-test assuming each population is at least 20 times larger than its respective sample.
n1 = 100
x1 = 308
s1 = 84

n2 = 100
x2 = 254
s2 = 67

se = np.sqrt(s1**2/n1 +s2**2/n2)
t_test = (x1-x2)/se
t_test

5.025702668336442

In [34]:
df = (s1**2/n1 + s2**2/n2)**2 / ( (s1**2 / n1)**2 / (n1 - 1)  + (s2**2 / n2)**2 / (n2 - 1) )
df

188.67235245796877

In [35]:
#Assuming alpha=0.05
alpha=0.05
p = 1-alpha/2
n = 16
t_score = stats.t.ppf(p,df)
t_score

1.9726171529778553

In [36]:
print("H0 rejected") if t_test > t_critical else print("H0 accepted")

H0 rejected


Problem Statement 13:
Pooled estimate of the population variance
Does the data provide sufficient evidence to conclude that average percentage
increase in the price of sugar differs when it is sold at two different prices?
Population 1: Price of sugar = Rs. 27.50 n1 = 14
x1 = 0.317%
s1 = 0.12%
Population 2: Price of sugar = Rs. 20.00 n2 = 9
x2 = 0.21%
s2 = 0.11%

H0 : mu1 = mu2
H1 : mu1 != mu2

In [29]:
#for pooled estimate of the common population variance:
s_2 = ((n1-1)*(s1**2)+(n2-1)*(s2**2)) / (n1+n2-2)
se = np.sqrt(s_2*(1/n1+1/n2))
t_test = (x1-x2)/se
t_test

5.025702668336442

In [30]:
alpha = 0.05
p = 1-alpha/2
df = n1+n2-2
t_critical = stats.t.ppf(p,df)
t_critical

1.9720174778338955

In [31]:
print("H0 rejected") if t_test > t_critical else print("H0 accepted")

H0 rejected


Problem Statement 14:
The manufacturers of compact disk players want to test whether a small price
reduction is enough to increase sales of their product. Is there evidence that the
small price reduction is enough to increase sales of compact disk players?
Population 1: Before reduction
n1 = 15
x1 = Rs. 6598 s1 = Rs. 844
Population 2: After reduction n2 = 12
x2 = RS. 6870
s2 = Rs. 669

H0 : mu1 = mu2
H1 : mu1 != mu2

In [38]:
n1 = 15 
x1 = 6598 
s1 = 844 
n2 = 12 
x2 = 6870 
s2 = 669

In [41]:
s_2 = ((n1-1)*(s1**2)+(n2-1)*(s2**2)) / (n1+n2-2)
se = np.sqrt(s_2*(1/n1+1/n2))
t_test = (x1-x2)/se
t_test

-0.9098300343990461

In [43]:
alpha = 0.05
p = 1-alpha/2
df = n1+n2-2
t_critical = stats.t.ppf(p,df)
t_critical

2.059538552753294

In [44]:
print("H0 rejected") if t_test > t_critical else print("H0 accepted")

H0 accepted


Problem Statement 15:
Comparisons of two population proportions when the hypothesized difference is zero
Carry out a two-tailed test of the equality of banks’ share of the car loan market in
1980 and 1995.
Population 1: 1980
n1 = 1000
x1 = 53
𝑝 1 = 0.53
Population 2: 1985
n2 = 100
x2 = 43
𝑝 2= 0.53

H0: p1-p2 = 0
H1: p1-p2!= 0

In [45]:
n1=100
x1 = 53
p_1 = 0.53
n2 = 100
x2 = 43
#p_2 = 0.53, Data provided #since we know, p_2 = x2/n2. 
#Data must have got some error so making change.
p_2 = 0.43

In [47]:
D=0
p = (x1+x2)/(n1+n2)
numerator = (p_1-p_2)-D
denominator = np.sqrt(p*(1-p)*(1/n1 + 1/n2))
z_test = numerator/denominator
z_test

1.4153462926807452

In [50]:
alpha = 0.05
p=1-alpha/2
z_criticial=stats.norm.ppf(p)
z_criticial

1.959963984540054

In [51]:
print("H0 rejected") if z_test > z_critical else print("H0 accepted")

H0 accepted


Problem Statement 16:
Carry out a one-tailed test to determine whether the population proportion of
traveler’s check buyers who buy at least $2500 in checks when sweepstakes prizes
are offered as at least 10% higher than the proportion of such buyers when no
sweepstakes are on.
Population 1: With sweepstakes
n1 = 300
x1 = 120
𝑝 = 0.40
Population 2: No sweepstakes n2 = 700
x2 = 140
𝑝 2= 0.20

H0: p1-p2 >= 0.1
H0: p1-p2 < 0.1
i.e. one tail test

In [58]:
#population 1 : With sweepstakes
n1=300
x1=120
p_1=0.40

#population 2: Without
n2=700
x2=140
p_2=0.20

D=0.1 #difference in population proportion

In [63]:
numerator = (p_1-p_2)-D
denominator = np.sqrt(p_1*(1-p_1)/n1 +p_2*(1-p_2)/n2)
z_test = numerator/denominator
z_test

3.1180478223116177

In [64]:
alpha = 0.01
p=1-alpha
z_critical=stats.norm.ppf(p)
z_critical

2.3263478740408408

In [65]:
print("H0 rejected") if z_test > z_critical else print("H0 accepted")

H0 rejected


Problem Statement 17:
A die is thrown 132 times with the following results: 
Number turned up: 1, 2, 3, 4, 5, 6
Frequency:       16, 20, 25, 14, 29, 28
Is the die unbiased? 
Consider the degrees of freedom as 𝑝_hat − 1

Chi-square for goodness of fit is when we have only one variable dice-rolled
H0: die are unbiased i.e. (O-E) = 0
H1 : die are biased i.e.  (O-E) != 0

In [90]:
# for die to be unbiased every number must come equal time
n=132
value = [1,2,3,4,5,6]
O =     [16,20,25,14,29,28]
E_val = n/len(value)
E = [E_val for _ in range(len(value))]
E

[22.0, 22.0, 22.0, 22.0, 22.0, 22.0]

In [91]:
chi_test = 0
for i in range(len(O)):
    chi_test += (O[i]-E[i])**2/E[i]
chi_test

9.0

In [92]:
alpha = 0.05
p= 1-alpha
df = len(O)-1
chi_critical=stats.chi2.ppf(p,df)
chi_critical

11.070497693516351

In [95]:
print("Hypothesis rejected as per chi score") if chi_test > chi_critical else print("Hypothesis accepted as per chi score")

Hypothesis accepted as per chi score


Problem Statement 18:
In a certain town, there are about one million eligible voters. A simple random
sample of 10,000 eligible voters was chosen to study the relationship between
gender and participation in the last election. The results are summarized in the
following 2X2 (read two by two) contingency table:
          Men    Women
Voted     2792   3591
Not Voted 1486   2131
We would want to check whether being a man or a woman (columns) is independent of
having voted in the last election (rows). In other words, is “gender and voting independent”?

Chisquare test for independence. It is similar to goodness of fit but here it has two variable Formula: 

Expected(row,column) = (row_total * column_total) / grand_total

H0: gender is gender and voting independent
H1: gender is gender and voting dependent

In [105]:
table =[[2792,3591],
       [1486,2131]]
table

[[2792, 3591], [1486, 2131]]

In [112]:
chi_test, prob, dof, expected = stats.chi2_contingency(table)
print('dof:{} and p:{}'.format(dof,prob))

dof:1 and p:0.01047464310091866


In [113]:
alpha = 0.05
p = 1-alpha
chi_critical = stats.chi2.ppf(p, dof)
chi_critical

3.841458820694124

In [111]:
print("Hypothesis rejected as per chi score") if chi_test > chi_critical else print("Hypothesis accepted as per chi score")

Hypothesis rejected as per chi score


Problem Statement 19:
A sample of 100 voters are asked which of four candidates they would vote for in an
election. The number supporting each candidate is given below:
Higgins Reardon White Charlton
41      19      24    16
Do the data suggest that all candidates are equally popular? 
[Chi-Square = 14.96, with 3 df, 𝑝 0.05 .

In [ ]:
Goodness of Fit
H0: candidates are equally popular
H1: candidates are not equally popular

In [116]:
total_voter = 100
vote_O = [41,19,24,16]
E_value = total_voter/len(vote_O)
vote_E = [E_value for _ in range(len(vote_O))]
vote_E

[25.0, 25.0, 25.0, 25.0]

In [117]:
chi_test = 0
for i in range(len(vote_O)):
    chi_test += (vote_O[i]-vote_E[i])**2/vote_E[i]
chi_test

14.959999999999999

In [118]:
df = len(vote_O)-1
alpha = 0.05
p=1-alpha
critical = stats.chi2.ppf(p,df)
critical

7.814727903251179

In [119]:
print("Hypothesis rejected as per chi score") if chi_test > chi_critical else print("Hypothesis accepted as per chi score")

Hypothesis rejected as per chi score


Problem Statement 20:
Children of three ages are asked to indicate their preference for three photographs of
adults. Do the data suggest that there is a significant relationship between age and
photograph preference? What is wrong with this study? 
[Chi-Square = 29.6, with 4 df: 𝑝 < 0.05].

                   Photograph
Age Of Child       A    B    C
5-6 Years          18   22   20
7-8 Years          2    28   40
9-10 Years         20   10   40

H0: no significant relationship between age and photograph
H1:significant relationship between age and photograph

In [121]:
table =[[18,22,20],
       [2,28,40],
       [20,10,40]]
table

[[18, 22, 20], [2, 28, 40], [20, 10, 40]]

In [122]:
chi_test, prob, dof, expected = stats.chi2_contingency(table)
print('dof:{} and p:{}'.format(dof,prob))

dof:4 and p:5.894582174262606e-06


In [123]:
alpha = 0.05
p = 1-alpha
chi_critical = stats.chi2.ppf(p, dof)
chi_critical

9.487729036781154

In [124]:
print("Hypothesis rejected as per chi score") if chi_test > chi_critical else print("Hypothesis accepted as per chi score")

Hypothesis rejected as per chi score


Problem Statement 21:
A study of conformity using the Asch paradigm involved two conditions: one where
one confederate supported the true judgement and another where no confederate
gave the correct response.
Is there a significant difference between the "support" and "no support" conditions in the
frequency with which individuals are likely to conform? 
[Chi-Square = 19.87, with 1 df: 𝑝 < 0.05].

             Support     No Support
Conform      18          40
Non conform  32          10

H0: no significant difference in support and no support
H1: signigicant difference in support and no support

In [125]:
table =[[18,40],
       [32,10]]
table

[[18, 40], [32, 10]]

In [126]:
chi_test, prob, dof, expected = stats.chi2_contingency(table)
print('dof:{} and p:{}'.format(dof,prob))

dof:1 and p:2.092221924065813e-05


In [127]:
alpha = 0.05
p = 1-alpha
chi_critical = stats.chi2.ppf(p, dof)
chi_critical

3.841458820694124

In [128]:
print("Hypothesis rejected as per chi score") if chi_test > chi_critical else print("Hypothesis accepted as per chi score")

Hypothesis rejected as per chi score


Problem Statement 22:
We want to test whether short people differ with respect to their leadership qualities
(Genghis Khan, Adolf Hitler and Napoleon were all stature-deprived, and how many midget
MP's are there?) The following table shows the frequencies with which 43 short people and
52 tall people were categorized as "leaders", "followers" or as "unclassifiable". Is there a
relationship between height and leadership qualities?
[Chi-Square = 10.71, with 2 df: 𝑝 < 0.01].

              Height
              Short   Tall
Leader        12      32
Follower      22      14
Unclassified  9       6

H0: no relationship in between height and leadership quality
H1: relationship in between height and leadership quality

In [130]:
table =[[12,32],
       [22,14],
       [9,6]]
table

[[12, 32], [22, 14], [9, 6]]

In [131]:
chi_test, prob, dof, expected = stats.chi2_contingency(table)
print('dof:{} and p:{}'.format(dof,prob))

dof:2 and p:0.004719280137040844


In [132]:
alpha = 0.05
p = 1-alpha
chi_critical = stats.chi2.ppf(p, dof)
chi_critical

5.991464547107979

In [133]:
print("Hypothesis rejected as per chi score") if chi_test > chi_critical else print("Hypothesis accepted as per chi score")

Hypothesis rejected as per chi score


Problem Statement 23:
Each respondent in the Current Population Survey of March 1993 was classified as
employed, unemployed, or outside the labor force. The results for men in California age 35-
44 can be cross-tabulated by marital status, as follows:

                     Married   (Widowed, divorced 
                                or seperated)       (Never married)  
Employed             679        103                 114
Unemployed           63         10                  20
Not in labor force   42         18                  25

Men of different marital status seem to have different distributions of labor force status. Or is
this just chance variation? (you may assume the table results from a simple random
sample.)

H0: no relationship between married status and employment status
H1: relationship between married status and employment status

In [134]:
table =[[679,103,114],
       [63,10,20],
       [42,18,25]]
table

[[679, 103, 114], [63, 10, 20], [42, 18, 25]]

In [135]:
chi_test, prob, dof, expected = stats.chi2_contingency(table)
print('dof:{} and p:{}'.format(dof,prob))

dof:4 and p:2.2949883042859918e-06


In [136]:
alpha = 0.05
p = 1-alpha
chi_critical = stats.chi2.ppf(p, dof)
chi_critical

9.487729036781154

In [137]:
print("Hypothesis rejected as per chi score") if chi_test > chi_critical else print("Hypothesis accepted as per chi score")

Hypothesis rejected as per chi score
